### PR TITLE
Refactor test_storage.py to use base test classes

### PR DESCRIPTION
**Pull request by Fenn-CS**
There are base test classes namely `SyncTestCase` and `AsyncTestCase` which we would like all test classes in this code base to extend.

This commit extends the listed classes in test_storage.py to extend the above mentioned base classes:

  * UtilTests
  * BucketProxy
  * Server

### CODE REVIEW

File: src/allmydata/test/test_storage.py [hunk 1]
```diff
@@ -25,8 +25,16 @@
 from functools import partial
 from uuid import uuid4
 
+from .common import (
+    SyncTestCase,
+    AsyncTestCase,
+)
```
Comment (exarkun) [inline src/allmydata/test/test_storage.py:31]: Looks like one of these was already imported below.

We should figure out how to [isort](https://pypi.org/project/isort/) the codebase, probably.

**Review by exarkun — CHANGES_REQUESTED** [review]
Thanks!  Basically looks good to me except one minor issue, comment inline.

**Comment by itamarst** [comment]
My branch definitely causing those test failures :(

File: src/allmydata/test/test_storage.py (hunk 1 above)
Comment (Fenn-CS) [reply src/allmydata/test/test_storage.py:31]: thanks, would sort it out.

**Comment by Fenn-CS** [comment]
Thanks for the initial review @exarkun this is now ready for another pass!

File: src/allmydata/test/test_storage.py [hunk 2]
```diff
@@ -1266,13 +1275,13 @@ def test_advise_corruption_missing(self):
...
-        self.assertEqual(
+        self.assertThat(
             [],
-            os.listdir(ss.corruption_advisory_dir),
+            Equals(os.listdir(ss.corruption_advisory_dir)),
         )
 
 
-class MutableServer(unittest.TestCase):
+class MutableServer(SyncTestCase):
```
Comment (exarkun) [inline src/allmydata/test/test_storage.py:1284]: setUp and tearDown need to call the super implementations

File: src/allmydata/test/test_storage.py [hunk 3]
```diff
@@ -1891,10 +1900,10 @@ def test_remove_non_present(self):
         )
 
         self.assertTrue(testv_is_good)
-        self.assertEqual({}, read_data)
+        self.assertThat({}, Equals(read_data))
 
 
-class MDMFProxies(unittest.TestCase, ShouldFailMixin):
+class MDMFProxies(SyncTestCase, ShouldFailMixin):
```
Comment (exarkun) [inline src/allmydata/test/test_storage.py:1906]: setUp and tearDown need to call the super implementations

File: src/allmydata/test/test_storage.py [hunk 4]
```diff
@@ -3309,21 +3318,21 @@ def _then(results):
...
-            self.failUnlessEqual(read(b"si1", [0], [(9, len(data) - 9)]),
-                                 {0: [data[9:]]})
+            self.assertThat(read(b"si1", [0], [(9, len(data) - 9)]),
+                                 Equals({0: [data[9:]]}))
         d.addCallback(_then_again)
         return d
 
 
-class Stats(unittest.TestCase):
+class Stats(SyncTestCase):
```
Comment (exarkun) [inline src/allmydata/test/test_storage.py:3335]: setUp and tearDown need to call the super implementations

**Review by exarkun — CHANGES_REQUESTED** [review]
Thanks!  Generally looking very good.  Some small changes suggested inline.

File: src/allmydata/test/test_storage.py [hunk 5]
```diff
@@ -2141,7 +2150,7 @@ def write_sdmf_share_to_server(self,
         tws[0] = (testvs, [(0, share)], None)
         readv = []
         results = write(storage_index, self.secrets, tws, readv)
-        self.failUnless(results[0])
+        self.assertFalse(results[0])
```
[the remark this answers is not in the document]
Comment (Fenn-CS) [reply src/allmydata/test/test_storage.py:2153]: Thanks resolved.

File: src/allmydata/test/test_storage.py [hunk 6]
```diff
@@ -430,7 +436,7 @@ def _call():
         return defer.maybeDeferred(_call)
 
 
-class BucketProxy(unittest.TestCase):
+class BucketProxy(SyncTestCase):
```
[the remark this answers is not in the document]
Comment (Fenn-CS) [reply src/allmydata/test/test_storage.py:439]: Now using `AsyncTestCase`

File: src/allmydata/test/test_storage.py [hunk 7]
```diff
@@ -2422,14 +2431,14 @@ def test_write_test_vectors(self):
...
             res, d = results
-            self.failIf(res)
+            self.assertFalse(res)
 
         def _check_success(results):
-            self.failUnlessEqual(len(results), 2)
+            self.assertThat(results, HasLength(2))
             res, d = results
-            self.failUnless(results)
+            self.assertFalse(results)
```
[the remark this answers is not in the document]
Comment (Fenn-CS) [reply src/allmydata/test/test_storage.py:2441]: Fixed, thanks

File: src/allmydata/test/test_storage.py [hunk 8]
```diff
@@ -2888,51 +2897,51 @@ def _check_block_and_salt(results):
...
-            self.failUnlessEqual(self.share_hash_chain,
-                                 sharehashes))
+            self.assertThat(self.share_hash_chain,
+                                 Equals(sharehashes)))
         #  - The keys
         d.addCallback(lambda ignored:
             mr.get_encprivkey())
         d.addCallback(lambda encprivkey:
-            self.failUnlessEqual(encprivkey, self.encprivkey, encprivkey))
+            self.assertThat(encprivkey, self.encprivkey, Equals(encprivkey)))
```
[the remark this answers is not in the document]
Comment (Fenn-CS) [reply src/allmydata/test/test_storage.py:2921]: Updated
```suggestion
            self.assertThat(encprivkey, Equals(self.encprivkey), encprivkey))
```

File: src/allmydata/test/test_storage.py (hunk 4 above)
Comment (Fenn-CS) [reply src/allmydata/test/test_storage.py:3335]: Resolved

File: src/allmydata/test/test_storage.py (hunk 3 above)
Comment (Fenn-CS) [reply src/allmydata/test/test_storage.py:1906]: Resolved by calling super implementations

File: src/allmydata/test/test_storage.py [hunk 9]
```diff
@@ -550,13 +556,16 @@ def test_readwrite_v2(self):
...
+class Server(AsyncTestCase):
 
     def setUp(self):
+        super(Server, self).setUp()
         self.sparent = LoggingServiceParent()
         self.sparent.startService()
         self._lease_secret = itertools.count()
+
     def tearDown(self):
+        super(Server, self).tearDown()
```
[the remark this answers is not in the document]
Comment (Fenn-CS) [reply src/allmydata/test/test_storage.py:568]: Thanks, went with `addCleanUp` for the one-liner `tearDowns` that were affected by these corrections.

**Comment by Fenn-CS** [comment]
Thanks @exarkun all your comments were addressed, please take another pass.

**Comment by exarkun** [comment]
> Thanks @exarkun all your comments were addressed, please take another pass.

Thanks.  In the future, it would be easier to re-review if you avoid force-pushing.  It's a little easier to see what has changed since last time without history rewriting being involved.

File: src/allmydata/test/test_storage.py [hunk 10]
```diff
@@ -563,10 +563,7 @@ def setUp(self):
         self.sparent = LoggingServiceParent()
         self.sparent.startService()
         self._lease_secret = itertools.count()
-
-    def tearDown(self):
-        super(Server, self).tearDown()
-        return self.sparent.stopService()
+        self.addCleanup(self.sparent.stopService())
```
[the remark this answers is not in the document]
Comment (Fenn-CS) [reply src/allmydata/test/test_storage.py]: Thanks @exarkun, this is now resolved.

**Comment by Fenn-CS** [comment]
> Looks like the issue I spotted wasn't the only trouble CI was having since a bunch of builds are still red. It looks like the changes for these tests need another look to see what went wrong:
> 
>     MDMFProxies.test_is_sdmf
>     MDMFProxies.test_write_test_vectors
>     MutableServer.test_remove
>     MDMFProxies.test_verinfo_with_mdmf_file
>     MutableServer.test_allocate


Looked through these methods and found some more inverted assertions so I fixed them! Thanks @exarkun, kindly re-review!

**Review by exarkun — APPROVED** [review]
Great, thank you!

On my earlier review I clearly missed some things.  I think this would be less likely to happen for a smaller changeset so if you want to split future refactorings of long test modules into multiple PRs I certainly would not object.

**Pull request merged**